packages/back-end/src/services/features.ts

```diff
@@ -89,6 +146,7 @@ export async function refreshSDKPayloadCache(
   );
   payloadKeys = payloadKeys.filter((k) => allowedEnvs.has(k.environment));
 
+  const allVisualExperiments = await getAllVisualExperiments(organization.id);
```

Comment:
Move this down a few lines. We have an early return if there are no payload keys (which happens frequently with feature updates) and there's no reason to do this db query if we're just going to immediately return.

Reply:
Sorry - I meant to ask about this, I think I forgot to put a TODO. Do we want to exit early if no payload keys are defined?  For example, will there be a situation where experiments need updating but not features? I was thinking of doing something like the following:

```
  const allVisualExperiments = await getAllVisualExperiments(organization.id);
  // If no environments are affected, we don't need to update anything
  if (!payloadKeys.length || !allVisualExperiments.length) return;
```

Reply:
Just to clarify, when an experiment is updated, we're also going to pass in affected payload keys into this function.  The way we calculate which payload keys are affected is going to be slightly different for experiments since there are no environments, but the end result should be the same.  We want to skip refreshing the cache if nothing meaningful changed

Reply:
Got it; updated